bokehjs/src/coffee/models/annotations/legend.coffee

```diff
@@ -88,8 +88,12 @@ export class LegendView extends AnnotationView
           y = (v_range.end + v_range.start)/2 + legend_height/2
     else if isArray(location) and location.length == 2
       [x, y] = location   # left, bottom wrt panel
-      x += h_range.start
-      y += v_range.start + legend_height
+      if panel.side?
```

Comment:
This is a likely to break in future. I would use `if panel.side in ['left', 'right', 'above', 'below']`.